src/operator/do.ts

```diff
@@ -4,14 +4,50 @@ import {Observable} from '../Observable';
 import {PartialObserver} from '../Observer';
 
 /**
- * Returns a mirrored Observable of the source Observable, but modified so that the provided Observer is called
- * for every item emitted by the source.
- * This operator is useful for debugging your observables for the correct values or performing other side effects.
- * @param {Observer|function} [nextOrObserver] a normal observer callback or callback for onNext.
- * @param {function} [error] callback for errors in the source.
- * @param {function} [complete] callback for the completion of the source.
- * @reurns {Observable} a mirrored Observable with the specified Observer or callback attached for each item.
+ * Perform a side effect for every emission on the source Observable, but return
+ * an Observable that is identical to the source.
+ *
+ * <span class="informal">Intercepts each emission on the source and runs a
+ * function, but returns an output which is identical to the source.</span>
+ *
+ * <img src="./img/do.png" width="100%">
+ *
+ * Returns a mirrored Observable of the source Observable, but modified so that
+ * the provided Observer is called for every item emitted by the source.
+ *
+ * This operator is useful for debugging your Observables for the correct values
+ * or performing other side effects.
```

Comment:
I'd swap this around to say it's useful for side effects *first*, and _then_ mention it's useful for debugging.